tools/nl/embeddings/build_embeddings_v2_test.py

```diff
@@ -137,3 +158,126 @@ def testSuccess(self):
+                                                    "palm_alternatives.csv")
+    output_dcid_sentences_filepath = os.path.join(
+        "csv", "embeddings_input_sv_descriptions.csv")
+
+    # Check that all the files exist.
+    self.assertTrue(os.path.exists(sheets_filepath))
+    self.assertTrue(os.path.exists(input_other_alternatives_filepath))
+    self.assertTrue(os.path.exists(input_palm_alternatives_filepath))
+    self.assertTrue(os.path.exists(output_dcid_sentences_filepath))
+
+    # Perform input file validations.
+
+    # The input data files should not have the same sentence/description map to
+    # different SV dcids.
+    sheets_df = pd.read_csv(sheets_filepath).fillna("")
+    expected_cols = [
+        "dcid", "Name", "Description", "Override_Alternatives",
+        "Curated_Alternatives"
+    ]
+    sheets_sentence_cols = expected_cols[1:]
+
+    self.assertCountEqual(expected_cols, sheets_df.columns.values)
+    _validate_sentence_to_dcid_map(self, sheets_df, "dcid",
+                                   sheets_sentence_cols)
+
+    palm_alts_df = pd.read_csv(input_palm_alternatives_filepath).fillna("")
+    other_alts_df = pd.read_csv(input_other_alternatives_filepath).fillna("")
+    expected_cols = ["dcid", "Alternatives"]
+    alt_sentence_cols = expected_cols[1:]
+
+    self.assertCountEqual(expected_cols, palm_alts_df.columns.values)
+    self.assertCountEqual(expected_cols, other_alts_df.columns.values)
+    _validate_sentence_to_dcid_map(self, palm_alts_df, "dcid",
+                                   alt_sentence_cols)
+    _validate_sentence_to_dcid_map(self, other_alts_df, "dcid",
+                                   alt_sentence_cols)
+
+  def testOutputFileValidations(self):
+    output_dcid_sentences_filepath = os.path.join(
+        "csv", "embeddings_input_sv_descriptions.csv")
+
+    dcid_sentence_df = pd.read_csv(output_dcid_sentences_filepath).fillna("")
+
+    # Check that the dcids do not have duplicate rows.
+    self.assertEqual(len(dcid_sentence_df),
+                     len(dcid_sentence_df["dcid"].unique()))
+    self.assertEqual(len(dcid_sentence_df),
+                     len(dcid_sentence_df["sentence"].unique()))
+
+    # Check that there are no duplicate sentences.
+    sentences = {}
+    for row_index, row in dcid_sentence_df.iterrows():
+      sv_dcid = row["dcid"]
+      for s in row["sentence"].split(";"):
+        if not s:
+          continue
+
+        existing_sv_index = sentences.get(s, ())
+        err_msg = f"\nSentence already exists: {s}.\nCurrent (SV, index): {sv_dcid, row_index}.\nExisting (SV, index): {existing_sv_index}."
+        self.assertFalse(s in sentences, err_msg)
+        sentences[s] = (sv_dcid, row_index)
+
+  def testEmbeddingsValidations(self):
+    # TODO: consider moving this test to a separate file which is
+    # tiggered on demand. This test can take a while to run (esp when the
+    # number of SVs and sentences becomes larger in the future.
```

Comment:
How long does this take now?  Does this test run on every website presubmit?

Reply:
About 30s. And yes, right now I think it will run every single time.

Reply:
I think we should perhaps fork this out for running when we update the index?